doc/source/serve/production.md

```diff
@@ -547,7 +556,14 @@ $ python
 
 The price has updated! The same request now returns `10` instead of `6`, reflecting the new price.
 
-You can update any setting in any deployment in the config file similarly. You can also add new deployment settings or remove old deployment settings from the config. This is because `serve deploy` is **idempotent**. Your Serve application's will match the one specified in the latest config you deployed– regardless of what config files you deployed before that.
+### Code Updates
+
+Similarly, you can update any other setting in any deployment in the config file. If a deployment setting other than `num_replicas`, `autoscaling_config`, or `user_config` is changed, it is considered a code update and the deployment replicas will be restarted. Note that the following are all considered "changes", and (if considered a code update) will trigger tear down of replicas:
+* Changing an existing setting.
+* Adding an override setting that was previously not present in the config file.
+* Removing a setting from the config file.
+
```

Comment:
If a deployment setting other than `num_replicas`, `autoscaling_config`, or `user_config` is changed, it is considered a code update, and the deployment replicas will be restarted. Note that the following modifications are all considered "changes," and (considered as a code update) will trigger tear down of replicas:
* changing an existing setting
* adding an override setting that was previously not present in the config file
* removing a setting from the config file